c3d/src/C3d.cpp

```diff
@@ -206,46 +360,28 @@ class C3dConstraintAttrInfo : public C3dSimpleSpelling {
       return consumeUntilClosingParenAndError(P);
     }
 
+    Sema &S = P->getActions();
+
     // Extend the scope so that we don't have resolution errors. We fake a
     // variable declaration just so that the name resolution is happy, and then
     // assume that the current context will be dropped anyhow.
     LLVM_DEBUG(llvm::dbgs() << "c3d: scope extension! " << D->getIdentifier()->getNameStart() << "\n");
-
-    // Check if we are accumulating another constraint on the same field, in
-    // which case no need to add this field to the scope
-    // TODO: this did not work out. Why?
-    // LookupResult LR(P->getActions(), D->getIdentifier(), D->getBeginLoc(), Sema::LookupAnyName);
-    // if (LR.getResultKind() != LookupResult::Found) { ... }
-    Sema &S = P->getActions();
-    bool InScope = false;
-    // TODO: figure out how to use specific_decl_iterator
-    // TODO: not all declarations have names
-    for (const auto &Decl: S.getCurScope()->decls())
-      if (const auto *VD = dyn_cast<VarDecl>(Decl))
-        if (VD->getName() == D->getIdentifier()->getName())
-          InScope = true;
+    // Check if we are accumulating another constraint on the same
+    // field, in which case no need to add this field to the scope. It
+    // can only be in the last posititon.
+    bool InScope =
+        !c3d_scope.empty() &&
+         (c3d_scope.back().Id->getName() == D->getIdentifier()->getName());
```

Comment:
this might be for cases where, e.g., a global variable with the same name is already in the clang scope, but we're happy to perform shadowing of it just for this particular field

Reply:
Exactly, for
```c
string foo;

struct
  [[everparse::process]]
  tt
{
    int foo [[everparse::constraint(foo <= 255)]];
};
```
We shadow the global `foo` with our own. This seems more correct actually since `sizeof foo` will now resolve to the size of the field instead of the global (though I may be confused about how the phases go).